sdk/core/core/inc/az_span.h

```diff
@@ -380,7 +380,7 @@ AZ_NODISCARD az_result az_span_append_dtoa(az_span destination, double source, a
  */
 AZ_INLINE void az_span_set(az_span destination, uint8_t fill)
 {
-  memset(az_span_ptr(destination), fill, az_span_capacity(destination));
+  memset(az_span_ptr(destination), fill, (size_t)az_span_capacity(destination));
```

Comment:
>size_t)az_span_capacity(destination)) [](start = 42, length = 37)

Can we also please add flag to the gates?

-Werror -Wall -Wextra -Wunused -Wuninitialized -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wlogical-op -Waggregate-return -Wfloat-equal

Reply:
@danieljurek , didn't we have all warnings turned on?

Reply:
@danieljurek , any objections if we add those?

Reply:
I have no objections. A long time ago we tried to enable all warnings but that PR got blocked and we never revisited it because of higher priorities. 

Are there equivalent warnings for MSVC and Clang that we'd want to turn on?

Reply:
@hihigupt, why do we want `-Waggregate-return`?

**-Waggregate-return**
_Warn if any functions that return structures or unions are defined or called. (In languages where you can return an array, this also elicits a warning.)_
(https://gcc.gnu.org/onlinedocs/gcc-3.4.6/gcc/Warning-Options.html)

Reply:
We can remove them which are not ok for core ... as some of them are only ok in AZURE RTOS .


---
In reply to: [393228236](https://github.com/Azure/azure-sdk-for-c/pull/465#discussion_r393228236) [](ancestors = 393228236)

Reply:
@TiejunMS what do think about these flags for core?

---
In reply to: [393230426](https://github.com/Azure/azure-sdk-for-c/pull/465#discussion_r393230426) [](ancestors = 393230426,393228236)

Reply:
We will not enable `-Waggregate-return` in this PR, because it would result in design changes - basically it means we can't return structs from functions, and that needs a bigger discussion.
And also I have to pass on `-Wconversion` - I think that, unfortunately, it is unreasonable (has bug) in GCC:  https://stackoverflow.com/questions/47543097/g-warning-conversion-to-uint16-t-from-int-may-alter-its-value : Basically, there is absolutely no way to make the following code compile without warnings:
```
short a = 1;
short b = 2;
a += b; // would complain even if you write "a = a + b"; or a = (short)((short)a + (short)b)
```

Reply:
For Azure RTOS compiler flags, some of them are collected from our customers. They are not acceptable of any compiler warnings from Azure RTOS. We did complained about the compiler flag `-Wconversion`. We could either explain to our customers that it is an issue of GCC and let them accept it or change our source code slightly to get GCC build clean. We chose the later one. Here is an example,
```
unsigned short a = 0xFFFF;
unsigned short b = 1;
a += b;
```
The result of `a` is 0. But it is not certain whether developer is aware of overflow or not. Rather than GCC compiler warning, some static analysis tools will give warning to such assignment. In Azure RTOS, we changed it like following for products verified by static analysis tool.
```
a = (unsigned short)((a + b) & 0xFFFF); // &0xFFFF implies that developer is aware of precision loss during cast. 
```

Since we never return struct for all Azure RTOS, we included this compiler flag "-Waggregate-return". There are no specific reasons here.

Reply:
Thank you, I will give it a try.